src/core/providers/ogr/qgsogrprovider.cpp

```diff
@@ -951,6 +951,11 @@ void QgsOgrProvider::addSubLayerDetailsToSubLayerList( int i, QgsOgrLayer *layer
   }
 }
 
+uint QgsOgrProvider::subLayerCount() const
+{
+  return subLayers().size();
```

Comment:
I think this would be more efficient to return layerCount directly -- otherwise there's a lot of extra work going on in the subLayers() function which isn't needed (e.g. determining geometry types for sub layers) 

```suggestion
  return layerCount();
```

Reply:
Ah yes sure. Fixed.